app/controllers/parties_controller.rb

```diff
@@ -1,27 +1,28 @@
-class PartiesController < ActionController::Base
-  before_action :create_movie, only: [:create]
+class PartiesController < ApplicationController
+  before_action :set_movie, only: [:create]
   def new
-    session[:movie] = {title: params[:title],
-      duration: params[:duration],
-      api_id: params[:api_id]
-    }
+    session[:movie] = { title: params[:title],
+                        duration: params[:duration],
+                        api_id: params[:api_id] }
+    @movie = session[:movie]
     @party = Party.new(duration: params[:duration])
   end
-  
+
   def create
-    # binding.pry
-    current_user.parties.create(party_params.merge(movie: @movie))
+    pc = PartyCreator.new(params, current_user, @movie)
+    pc.make_the_party
```

Comment:
This is because PartyCreator is a new class that creates parties and a movie when that is needed...? Likely as I read more this will make more sense.

Reply:
Yeah When passing :guests as a form_with param, trying to create the party in the controller with strong params failed because :guests was not permitted, so sending the data to a poro that can handle the data better allows us to bypass that restriction.